### DISCUSSION TITLE
Run migrations from inside hooks?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#3926] [Q&A]
I'm exploring a plugin architecture where the plugin is a pb_hooks file that includes:

* install (run migrations to create collections, insert data, etc)
* uninstall (reverse the process)
* api endpoints

Is it possible to do things like `const collection = new Collection({...}); Dao(db).saveCollection(collection);` from within a pb_hook?

**Comment by ganigeorgiev** [comment]
You can create collections programmatically with the JS hooks but it is not possible at the moment to trigger the migration runner from the JS hooks. Aka. you'll need to do checks like `if (collectionNotExists) { create() }` on each app start.

Also note that it may not be possible to "uninstall/revert" a hook during runtime (eg. once routes are registered they cannot be removed at the moment).
